pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,434 @@
+                                    this.setState({ message: "Reporting was canceled", });
+                                    // falls through
+                                case TaskState.ERROR:
+                                    this.setState({ problemState: ProblemState.REPORTABLE, });
+                                    break;
+                                case TaskState.COMPLETED:
+                                    this.setState({ problemState: ProblemState.REPORTED, });
+                                    break;
+                                default:
+                                    break;
+                                }
+
+                                this.updateStatusFromBus();
+                            });
+                            task_proxy.addEventListener("Prompt", (event, object_path, message, type) => {
+                                this.setState({ message: "Waiting for input…" });
+                                const task_prompt = client.proxy("org.freedesktop.reportd.Task.Prompt", object_path);
+                                const props = {
+                                    body: message,
+                                };
+                                const footerProps = {
+                                    actions: [],
+                                    cancel_clicked: () => {
+                                        task_proxy.Cancel();
+                                    },
+                                };
+
+                                switch (type) {
+                                case PromptType.ASK:
+                                case PromptType.ASK_PASSWORD:
+                                    props.body = (
+                                        <div>
+                                            <p>{message}</p>
+                                            <input className="full-width"
+                                                   ref={(input) => { this.input = input }}
+                                                   type={type == PromptType.ASK_PASSWORD && "password"} />
+                                        </div>
+                                    );
+                                    footerProps.actions.push(
+                                        {
+                                            caption: "Send",
+                                            clicked: () => {
+                                                return task_prompt.wait().done(() => {
+                                                    task_prompt.Input = this.input.value;
+                                                    task_prompt.Commit();
+                                                });
+                                            },
+                                            style: "primary",
+                                        }
+                                    );
+                                    break;
```

Comment:
are `ASK_YES_NO_YESFOREVER` and `ASK_YES_NO_SAVE` handled somehow?

Reply:
No (at least not currently). I decided to not expose the “remember choice” functionality, since it might need additional bells and whistles to manage the stored configuration.